src/ec/packages/ec-component-site-header/site-header.html.twig

```diff
@@ -0,0 +1,127 @@
+          code: (string) (default: ''): Switcher language code, eg. 'en', 'fr', etc.
+        },
+    - "header_link" (associative array) (default: predefined structure): Link settings for header link. format:
+        {
+          url: (string) (default: ''): URL for link
+          aria_label: (string) (default: ''): Aria-label attribute value
+        },
+    - "header_image" (associative array) (default: predefined structure): Heading image settings. format:
+        {
+          src: (string) (default: ''): URL for image
+          alt: (string) (default: ''): Alt attribute for image
+          title: (string) (default: ''): Title attribute for image
+        }
+    - "icon_file_path" (string) (default: ''): URL to icons file
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: []): format: [
+        {
+          "name" (string) (default: ''),
+          "value" (string) (default: '')
+        },
+        ...
+      ]
+#}
+
+{# Default variables #}
+{% set _search_form = {
+  text_input: {
+    id: 'input-search',
+    name: 'search',
+    extra_classes: 'ecl-search-form__text-input'
+  },
+  button: {
+    variant: 'search',
+    icon: {
+      type: 'general',
+      name: 'search',
+      path: _icon_file_path,
+    },
+    label: 'Search',
+    extra_classes: 'ecl-search-form__button'
+  }
+}
+%}
+
+{% set _language = {
+  url: '',
+  label: '',
+  code: ''
+}
+%}
+
+{% set _header_link = {
+  url: '',
+  aria_label: ''
+}
+%}
+
+{% set _header_image = {
+  src: '',
+  alt: '',
+  title: '',
+}
+%}
+
+{% set _css_class = 'ecl-site-header' %}
+{% set _extra_attributes = '' %}
+{% set _icon_file_path = icon_file_path|default('') %}
+
+{# Merge options #}
+{% if language is defined and language is not empty %}
+  {% set _language = _language|merge(language) %}
+{% endif %}
+
+{% if header_link is defined and header_link is not empty %}
+  {% set _header_link = _header_link|merge(header_link) %}
+{% endif %}
+
+{% if header_image is defined and header_image is not empty %}
+  {% set _header_image = _header_image|merge(header_image) %}
+{% endif %}
+
+{% if search_form.text_input is defined and search_form.text_input is not empty %}
+  {% set _search_form = _search_form|merge({ text_input: _search_form.text_input|merge(search_form.text_input) }) %}
+{% endif %}
+
+{% if search_form.button is defined and search_form.button is not empty %}
+  {% set _search_form = _search_form|merge({ button: _search_form.button|merge(search_form.button) }) %}
+{% endif %}
+
+{# Internal logic - Process properties #}
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+
```

Comment:
as discussed on slack yesterday, please add `{% spaceless %}` here (and `{% endspaceless %}` after the `</header>`)

Also every `{{` and `{%` in the spaceless area should be turned to `{{-` and `{%-` (same for the closing ones)

http://vvv.tobiassjosten.net/symfony/trim-and-remove-twig-whitespace/

Reply:
Check it in https://github.com/ec-europa/ecl-twig/pull/33/commits/db86360dccd7fd6ec6e71d44a13560fda4930945